lib/jsdom/browser/htmltodom.js

```diff
@@ -47,61 +45,27 @@ class HtmlToDom {
     return this["_parseWith" + this.parserType](html, false, element);
   }
 
-  _parseWithhtmlparser2(html, fragment, element) {
-    const handler = new this.parser.DefaultHandler();
-    // Check if document is XML
-    const isXML = this.parsingMode === "xml";
-    const parserInstance = new this.parser.Parser(handler, {
-      xmlMode: isXML,
-      lowerCaseTags: !isXML,
-      lowerCaseAttributeNames: !isXML,
-      decodeEntities: true
-    });
-
-    parserInstance.includeLocation = false;
-    parserInstance.parseComplete(html);
-
-    const parsed = handler.dom;
-    for (let i = 0; i < parsed.length; i++) {
-      setChild(this.core, element, parsed[i]);
-    }
-
-    return element;
-  }
-
-  _parseWithparse5v1(html, fragment, element) {
+  _parseWithparse5(html, fragment, element) {
     if (this.parsingMode === "xml") {
       throw new Error("Can't parse XML with parse5, please use htmlparser2 instead.");
     }
 
-    const htmlparser2Adapter = this.parser.TreeAdapters.htmlparser2;
-    let dom;
     if (fragment) {
-      const instance = new this.parser.Parser(htmlparser2Adapter);
-      const parentElement = htmlparser2Adapter.createElement(element.tagName.toLowerCase(), element.namespaceURI, []);
-      dom = instance.parseFragment(html, parentElement);
+      const domFragment = this.parser.parseFragment(element, html, {
+        treeAdapter: adapter(element.ownerDocument)
+      });
+      element.appendChild(domFragment);
     } else {
-      const instance = new this.parser.Parser(htmlparser2Adapter, { locationInfo: true });
-      dom = instance.parse(html);
-    }
-
-    const parsed = dom.children;
-    for (let i = 0; i < parsed.length; i++) {
-      setChild(this.core, element, parsed[i]);
+      const instance = new this.parser.ParserStream({
+        treeAdapter: adapter(element),
+        locationInfo: true
```

Comment:
I would recommend to not enable `locationInfo` by default. It has significant impact on performance and since it's not commonly used, I suggest to make it optional and disabled by default in jsdom.